pyqtorch/primitive.py

```diff
@@ -7,7 +7,7 @@
 
 from pyqtorch.apply import apply_operator
 from pyqtorch.matrices import OPERATIONS_DICT, _controlled, _dagger
-from pyqtorch.utils import Operator, State, product_state
+from pyqtorch.utils import product_state
```

Comment:
Your circular import comes from here because you import `utils` which itself imports `primitive`.

Reply:
Resolved.